### PR TITLE
Add prettier to server

### PR DESCRIPTION
**Pull request by jikim-msft**
### Description

Board: https://dev.azure.com/fluidframework/internal/_workitems/edit/2118/

This PR allows running prettier in each sub-directory of release group and packages. Instead of adding the script to the root directory, developers can run `npm format` in any directory they are currently working on which will fix entire files accordingly to the prettier configuration.

There should be a separate PR to run prettier in Server directory.

### CODE REVIEW

**Review by Josmithr — CHANGES_REQUESTED** [review]
We'll also want to add the `format` script to the root package.json files in the `gitrest`, `historian` and `routerlicious` directories, since each of those manage multiple packages via lerna.

These scripts should invoke `lerna run format` like we do at the root of the azure release group to run `format` in each package under the directory.

This release group is a bit strange, in that it doesn't have a single root package.json, but has sub-directory groups. But we still want to make sure each of those sub-groups works as expected with regards to formatting.

**Comment by jikim-msft** [comment]
> We'll also want to add the `format` script to the root package.json files in the `gitrest`, `historian` and `routerlicious` directories, since each of those manage multiple packages via lerna.
> 
> These scripts should invoke `lerna run format` like we do at the root of the azure release group to run `format` in each package under the directory.
> 
> This release group is a bit strange, in that it doesn't have a single root package.json, but has sub-directory groups. But we still want to make sure each of those sub-groups works as expected with regards to formatting.

I missed adding the format script in each root directory. Thanks for the call! I made the changes and checked that the `npm run format` works in each root directory. As said in the PR description, I will create a separate PR running the prettier in server directory.

File: server/historian/package.json [hunk 1]
```diff
@@ -15,7 +15,9 @@
     "build:compile": "lerna run build:compile --stream",
     "build:genver": "lerna run build:genver --stream --parallel",
     "clean": "rimraf dist *.tsbuildinfo *.build.log",
+    "format": "lerna run format --no-sort --stream -- -- -- --color",
     "postinstall": "npm run postinstall:lerna",
+    "lerna": "lerna",
```
Comment (Josmithr) [inline server/historian/package.json:20]: This shouldn't be needed. Was something not working without this?

Comment (Josmithr) [reply server/historian/package.json:20]: Looks like this was added to a couple other package.json files. We'll want to remove them there too.

**Review by Josmithr — CHANGES_REQUESTED** [review]

**Comment by hedasilv** [comment]
Thanks for working on this! Does this change have any impact in day-to-day development on the server packages? In other words, should the folks who work on the server side (us) be aware of any change in the process besides being able to run `npm format` on server folders?

**Comment by Josmithr** [comment]
> Thanks for working on this! Does this change have any impact in day-to-day development on the server packages? In other words, should the folks who work on the server side (us) be aware of any change in the process besides being able to run `npm format` on server folders?

No workflow changes yet. We're setting things up to enable prettier repo-wide, this is just step one of getting the infra set up to support that.

At a later date, we will enable prettier as our default formatter with linter verification. At that point, devs will either need to enable format-on-save (or something similar), or run the "format" script before merging PRs.

We will make a more central announcement about this as we get closer to that date.

For now, the scripts will be available, but there are no requirements around their use.

File: server/historian/package.json [hunk 2]
```diff
@@ -15,8 +15,9 @@
     "build:compile": "lerna run build:compile --stream",
     "build:genver": "lerna run build:genver --stream --parallel",
     "clean": "rimraf dist *.tsbuildinfo *.build.log",
+    "format": "lerna run format --no-sort --stream -- -- -- --color",
     "postinstall": "npm run postinstall:lerna",
-    "lint": "lerna run lint --no-sort --stream",
+    "lerna": "lerna",
```
Comment (Josmithr) [inline server/historian/package.json:20]: Looks like the "lint" script was removed in this one, rather than the "lerna" script.

**Review by Josmithr — CHANGES_REQUESTED** [review]

**Comment by hedasilv** [comment]
> > Thanks for working on this! Does this change have any impact in day-to-day development on the server packages? In other words, should the folks who work on the server side (us) be aware of any change in the process besides being able to run `npm format` on server folders?
> 
> No workflow changes yet. We're setting things up to enable prettier repo-wide, this is just step one of getting the infra set up to support that.
> 
> At a later date, we will enable prettier as our default formatter with linter verification. At that point, devs will either need to enable format-on-save (or something similar), or run the "format" script before merging PRs.
> 
> We will make a more central announcement about this as we get closer to that date.
> 
> For now, the scripts will be available, but there are no requirements around their use.

Sounds great! Thanks for the detailed response!

**Review by Josmithr — APPROVED** [review]
Looks great! Thank you!!

**Comment by jikim-msft** [comment]
> Looks great! Thank you!!

I appreciate your reviews (once again 😉)!

**Pull request merged**